src/sync/Synchroniser.js

```diff
@@ -173,26 +173,37 @@ export class Synchroniser {
   push = async () => {
     this.setProgressMessage('Pushing changes to the server');
     this.setProgress(0);
-    let recordsToSync;
     let translatedRecords;
     this.setTotal(this.syncQueue.length);
     while (this.syncQueue.length > 0) {
-      recordsToSync = this.syncQueue.next(BATCH_SIZE);
-      translatedRecords = recordsToSync.map(record =>
-        generateSyncJson(this.database, this.settings, record)
-      );
+      const recordsToSync = this.syncQueue.next(BATCH_SIZE);
+      translatedRecords = [];
+      recordsToSync.forEach(translateRecord);
       await this.pushRecords(translatedRecords);
+      // Records that threw errors in translateRecord() are still removed
       this.syncQueue.use(recordsToSync);
       this.incrementProgress(recordsToSync.length);
     }
-  }
+
+    const translateRecord = (record) => {
+      try {
+        const translatedRecord = generateSyncJson(this.database, this.settings, record);
+        // If an error was thrown no record is pushed
+        translatedRecords.push(translatedRecord);
+      } catch (error) {
+        // If error not safe to continue sync throw it again to pass it up to next handler
+        // See outgoingSyncUtils.js
+        if (!error.isSafe) throw error;
```

Comment:
If error is thrown here sync will stop and this.syncQyeue.use will not be called, causing BugSnag notifications every 10mins ?

Reply:
Hmmm

Reply:
It probably will, but bugsnag should just treat it as 1 _error_ with many _events_:

![image](https://user-images.githubusercontent.com/7684221/36129703-5e1abca8-10cd-11e8-87d2-f91ffc370e4f.png)

Reply:
I don't think we should be sending the same error, every 10 minutes to bug snag. Maybe sync out Q should have a 'broken' boolean, then instead of deleting we can flag all records that cause an issue with 'broken'.  In this case we don't need safeGet method at all